impeller/renderer/render_pass.cc

```diff
@@ -10,9 +10,18 @@ RenderPass::RenderPass(std::weak_ptr<const Context> context,
                        const RenderTarget& target)
     : context_(std::move(context)),
       render_target_(target),
-      transients_buffer_(HostBuffer::Create()) {}
+      transients_buffer_() {
+  auto strong_context = context_.lock();
+  FML_DCHECK(strong_context);
+  transients_buffer_ = strong_context->GetHostBufferPool().Grab();
+}
 
-RenderPass::~RenderPass() = default;
+RenderPass::~RenderPass() {
```

Comment:
In the context of multi-threaded encoding, are we still always guaranteed to destruct the RenderPass on the raster thread, or could it get destructed on a worker thread?

If multiple threads can access this don't we need to mutex guard it? Should we mutex guard it just in case (tm) ?

Reply:
I don't think we have to worry about that.  The `RenderPass` lives for a limited time inside of EntityPass::Render.  So that should be all the same thread.

If we did have a situation where this could be used by multiple threads we'd probably want a lockless ring buffer instead of using a mutex.

Reply:
If you look in command_buffer_mtl.mm, for example, the task closure captures the render pass, so it may/will be destroyed on the worker thread. You might be winning any races locally.

Reply:
Yep, you're right.  I'll rework this.  What is the max number of these?  The number of swapchain images?

Reply:
Do you mean render passes or task threads?

Reply:
Render passes

Reply:
There is no limit, we'll have one render pass for the entity pass and one render pass for each offscreen texture. So if you draw 50x shapes under a color filter we'll have 51 render passes.

Reply:
Done, mutex added.